components/eam/src/chemistry/yaml/aero_model/f90_yaml/modal_aero_bcscavcoef_init_end_yml.f90

```diff
@@ -0,0 +1,38 @@
+#ifdef YAML_AERO_MODEL
+  ! YAML file output generation code- DO NOT PORT to C++
+  if(yaml%flag_print) then ! if this column exists in lchnk
+
+     !write output header
+     call write_output_header(unit_output)
+
+     !start writing data
+     !<add code for writing data here>
+     !call write_1d_output_var(unit_output, fld_name, dim, field, inp_out_str) !writes 1D variables of any dimension in the output python module
+     call write_2d_output_var(unit_output, 'scavimptblnum', size(scavimptblnum,1), ntot_amode, scavimptblnum) !writes 2D variables of any dimension in the output python module
```

Comment:
`scavimptblnum` is defined [here](https://github.com/eagles-project/e3sm_mam4_refactor/blob/26913eb0ffd0fb42cbaa0613f4ea0acec6524d93/components/eam/src/chemistry/modal_aero/aero_model.F90#L69) as scavimptblnum(nimptblgrow_mind:nimptblgrow_maxd, ntot_amode) where nimptblgrow_mind is -7. But in write_2d_output_var indexing starts in 1. Is this an issue?

Reply:
The two variables `scavimptblnum` and `scavimptblvol` have dimensions of (-7:12, ntot_amode) where -7:12 includes 0 (so the total length is 20, to answer your question in slack). Not sure how this type of indexing is treated in C++ but it could be an issue in testing file. One possible way is to change the dimensions of variables to 1:20 and revise the index accordingly (e.g., change `scavimptblvol(0,imode)` to `scavimptblvol(1-nimptblgrow_mind, imode)` ). @singhbalwinder do you have any suggestion on this issue?

Reply:
This is an interesting case, we need to add this capability since Fortran allows negative indices. Functions like `ubound` and `lbound` might be helpful here to identify the lower and upper bounds of an array within the `write_2d_output_var` subroutine.

Reply:
in C++, we used 0 to 19 for the index.